File: plugins/core/src/main/java/eu/geekhome/coreplugin/CorePlugin.java
package eu.geekhome.coreplugin;

import eu.geekhome.services.extensibility.PluginMetadata;
import eu.geekhome.services.localization.Resource;

import org.pf4j.PluginWrapper;
import org.pf4j.Plugin;

public class CorePlugin extends Plugin implements PluginMetadata {

    public CorePlugin(PluginWrapper wrapper) {
        super(wrapper);
    }

    @Override
    public void start() {
        System.out.println("WelcomePlugin.start()");
    }

    @Override
    public void stop() {
        System.out.println("WelcomePlugin.stop()");
    }

    @Override
    public Resource getName() {
        return R.plugin_name;
    }

    @Override
    public Resource getDescription() {
        return R.plugin_description;
    }
}

File: plugins/core/src/main/java/eu/geekhome/coreplugin/CorePlugin.kt
package eu.geekhome.coreplugin

import eu.geekhome.services.configurable.SettingGroup
import eu.geekhome.services.extensibility.PluginMetadata
import eu.geekhome.services.localization.Resource
import org.pf4j.Plugin
import org.pf4j.PluginWrapper

class CorePlugin(wrapper: PluginWrapper?) : Plugin(wrapper), PluginMetadata {
    override fun start() {
        println("WelcomePlugin.start()")
    }

    override fun stop() {
        println("WelcomePlugin.stop()")
    }

    override val name: Resource = R.plugin_name
    override val description: Resource = R.plugin_description
    override val settingGroups: List<SettingGroup>
        get() = listOf(LanDiscoverySettings())
}